src/message.rs

```diff
@@ -319,6 +319,11 @@ pub struct Message {
     pub(crate) text: Option<String>,
     pub(crate) subject: String,
     pub(crate) rfc724_mid: String,
+
+    // In-Reply-To header contents.
+    //
+    // It is not necessarily well-formed and may contain garbage for incoming messages. Parse it
+    // with `crate::mimeparser::parse_message_id()` rather than using directly.
```

Comment:
This is kind of vague.  Even with the test I'm not sure what went wrong.  Is it not possible to turn it into something well-formed when creating this struct?

Reply:
In the test starting whitespace after newline and before the actual value went into this field, as a result the quoted message was not found in the database.

It is possible to cleanup the value before storing, but previously core stored raw header values into the database, which may contain multiple addresses, groups etc., any sort of syntax. So I did not change database schema but started interpreting it correctly. Otherwise you need a migration to cleanup previously stored values.

Reply:
I didn't really mean changing the database.  Makes sense to not interpret things that are stored so bugs can be fixed in the future.

But this could be done in `.load_from_db()`  Or if you'd rather do it lazily you could make the field private and instead use an accessor function which does this.  Those approaches feel less error prone to me.

Reply:
I moved `parse_message_id` call to `load_from_db`.